### PR TITLE
Replace deprecated time val

### PR DESCRIPTION
**Pull request by jeremypw**
In preparation for 2038 ...

Replaced `time_t` types with `int64` and use `GLib.DateTime` instead of `GLib.TimeVal` and `GLib.Time`

No regressions found so far browsing my existing photos and videos but needs thorough testing for corner cases, time-zone effects etc.

### CODE REVIEW

File: src/VideoMetadata.vala [hunk 1]
```diff
@@ -76,7 +76,7 @@ private class QuickTimeMetadataLoader {
     }
 
     public MetadataDateTime? get_creation_date_time () {
-        return new MetadataDateTime ((time_t) get_creation_date_time_for_quicktime ());
+        return new MetadataDateTime ((int64) get_creation_date_time_for_quicktime ());
```
Comment (tintou) [inline src/VideoMetadata.vala:79]:
```suggestion
        return new MetadataDateTime (get_creation_date_time_for_quicktime ());
```

File: src/VideoMetadata.vala [hunk 2]
```diff
@@ -320,7 +320,7 @@ private class AVIMetadataLoader {
     }
 
     public MetadataDateTime? get_creation_date_time () {
-        return new MetadataDateTime ((time_t) get_creation_date_time_for_avi ());
+        return new MetadataDateTime ((int64) get_creation_date_time_for_avi ());
```
Comment (tintou) [inline src/VideoMetadata.vala:323]:
```suggestion
        return new MetadataDateTime (get_creation_date_time_for_avi ());
```

File: src/VideoSupport.vala [hunk 3]
```diff
@@ -164,7 +164,7 @@ public class VideoReader {
 
         if (exposure_time == 0) {
             // Use time reported by Gstreamer, if available.
-            exposure_time = (time_t) (reader.timestamp != null ?
+            exposure_time = (int64) (reader.timestamp != null ?
```
Comment (tintou) [inline src/VideoSupport.vala:167]:
```suggestion
            exposure_time = (reader.timestamp != null ?
```

File: src/VideoMetadata.vala (hunk 1 above)
Comment (jeremypw) [reply src/VideoMetadata.vala]: Done

File: src/VideoMetadata.vala (hunk 2 above)
Comment (jeremypw) [reply src/VideoMetadata.vala]: Done

File: src/VideoSupport.vala (hunk 3 above)
Comment (jeremypw) [reply src/VideoSupport.vala]: Done

File: src/Dialogs/AdjustDateTimeDialog.vala [hunk 4]
```diff
@@ -200,31 +199,26 @@ public class AdjustDateTimeDialog : Granite.Dialog {
...
             return;
 
-        original_time_label.set_text (_ ("Original: ") +
-                                      Time.local (original_time).format (use_24_hr_format ? _ ("%m/%d/%Y, %H:%M:%S") :
-                                              _ ("%m/%d/%Y, %I:%M:%S %p")));
+        original_time_label.set_text (
+            _ ("Original: ") +
+            (new DateTime.from_unix_local (original_time)).format (
+                use_24_hr_format ? _ ("%m/%d/%Y, %H:%M:%S") : _ ("%m/%d/%Y, %I:%M:%S %p")
+            )
```
Comment (tintou) [inline src/Dialogs/AdjustDateTimeDialog.vala:206]: Not really related with your PR but something like this is better for the i10n:
```suggestion
            _ ("Original: %s").printf (
                (new DateTime.from_unix_local (original_time)).format (
                    use_24_hr_format ? _ ("%m/%d/%Y, %H:%M:%S") : _ ("%m/%d/%Y, %I:%M:%S %p")
                )
            )
```

File: src/Dialogs/Dialogs.vala [hunk 5]
```diff
@@ -250,7 +250,7 @@ public string create_result_report_from_manifest (ImportManifest manifest) {
     StringBuilder builder = new StringBuilder ();
 
     string header = _ ("Import Results Report") + " (Photos " + Resources.APP_VERSION + " @ " +
-                    TimeVal ().to_iso8601 () + ")\n\n";
+                    (new DateTime.now_local ()).to_string () + ")\n\n";
```
Comment (tintou) [inline src/Dialogs/Dialogs.vala:253]:
```suggestion
                    (new DateTime.now_local ()).format_iso8601 () + ")\n\n";
```

File: src/sidebar/metadata/BasicProperties.vala [hunk 6]
```diff
@@ -262,10 +262,10 @@ private class BasicProperties : Properties {
...
 
         if (start_time != 0) {
-            string start_date = get_prettyprint_date (Time.local (start_time));
-            string start_time = get_prettyprint_time (Time.local (start_time));
-            string end_date = get_prettyprint_date (Time.local (end_time));
-            string end_time = get_prettyprint_time (Time.local (end_time));
+            string start_date = get_prettyprint_date (new DateTime.from_unix_local (start_time));
+            string start_time = get_prettyprint_time (new DateTime.from_unix_local (start_time));
+            string end_date = get_prettyprint_date (new DateTime.from_unix_local (end_time));
+            string end_time = get_prettyprint_time (new DateTime.from_unix_local (end_time));
```
Comment (tintou) [inline src/sidebar/metadata/BasicProperties.vala:268]:
```suggestion
            var start_dt = new DateTime.from_unix_local (start_time);
            string start_date = get_prettyprint_date (start_dt);
            string start_time = get_prettyprint_time (start_dt);
            var end_dt = new DateTime.from_unix_local (end_time);
            string end_date = get_prettyprint_date (end_dt);
            string end_time = get_prettyprint_time (end_dt);
```

File: src/sidebar/metadata/ExtendedProperties.vala [hunk 7]
```diff
@@ -93,9 +93,9 @@ private class ExtendedProperties : Properties {
             is_raw = (photo.get_master_file_format () == PhotoFileFormat.RAW);
             artist = metadata.get_artist ();
             copyright = metadata.get_copyright ();
-            time_t exposure_time_obj = metadata.get_exposure_date_time ().get_timestamp ();
-            exposure_date = get_prettyprint_date (Time.local (exposure_time_obj));
-            exposure_time = get_prettyprint_time_with_seconds (Time.local (exposure_time_obj));
+            int64 exposure_time_obj = metadata.get_exposure_date_time ().get_timestamp ();
+            exposure_date = get_prettyprint_date (new DateTime.from_unix_local (exposure_time_obj));
+            exposure_time = get_prettyprint_time_with_seconds (new DateTime.from_unix_local (exposure_time_obj));
```
Comment (tintou) [inline src/sidebar/metadata/ExtendedProperties.vala:98]:
```suggestion
            var exposure_dt = new DateTime.from_unix_local (exposure_time_obj);
            exposure_date = get_prettyprint_date (exposure_dt);
            exposure_time = get_prettyprint_time_with_seconds (exposure_dt);
```

File: src/Dialogs/AdjustDateTimeDialog.vala (hunk 4 above)
Comment (jeremypw) [reply src/Dialogs/AdjustDateTimeDialog.vala:206]: Yeah, I was studiously ignoring the myriad other coding issues :wink:

**Comment by jeremypw** [comment]
@tintou Thanks for the review an helpful suggestions, which I have incorporated.

**Review by tintou — APPROVED** [review]
